src/bin/cargo-plumbing/plumbing/locate_project.rs

```diff
@@ -0,0 +1,108 @@
+use std::fmt::Debug;
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use cargo_plumbing::CargoResult;
+use clap::ValueEnum;
+use serde::Serialize;
+use toml::Table;
+
+#[derive(Debug, Clone, Copy, ValueEnum)]
+enum MessageFormat {
+    Json,
+    Plain,
+}
+
+#[derive(Debug, clap::Args)]
+pub(crate) struct Args {
+    /// Locate the workspace root `Cargo.toml` instead of the current package
+    #[arg(long)]
+    workspace: bool,
+
+    /// The representation in which to print the project location
+    #[arg(long, value_enum, default_value_t = MessageFormat::Json)]
+    message_format: MessageFormat,
+
+    /// Path to the `Cargo.toml` file to start searching from
+    #[arg(long)]
+    manifest_path: Option<PathBuf>,
+}
+
+#[derive(Serialize)]
+struct ProjectLocation<'a> {
+    root: &'a str,
+}
```

Comment:
The end-goal this is to end up in Cargo and it would help to follow Cargo's layout (and Cargo's layout has some intention behind it).

Cargo is generally split up by
- `schemas` (serde types)
- `core` (generic functionality)
- `ops` (command-specific functionality)
- `bin` (CLI facade over `ops`)

Depending on our use of Cargo's infrastucture, there might not need to be an `ops` or `core`.

Cargo has a distinct `schemas`
- To separate out the compatibility guarantees as most Cargo structs have none while these do
- To allow for reuse

Hmm, to especially help with kicking the tires of this command, I wonder if we should fully split schemas out into a `cargo-plumbing-schemas` library.

Reply:
If you'd like